pilot/pkg/networking/core/cluster.go

```diff
@@ -70,6 +70,12 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(proxy *model.Proxy, req *mod
 func (configgen *ConfigGeneratorImpl) BuildDeltaClusters(proxy *model.Proxy, updates *model.PushRequest,
 	watched *model.WatchedResource,
 ) ([]*discovery.Resource, []string, model.XdsLogDetails, bool) {
+	// If FilterGatewayClusterConfig is enabled, we need to generate subset of clusters for the gateway.
+	if features.FilterGatewayClusterConfig && proxy.Type == model.Router {
```

Comment:
I'm curious to understand more about how this works... are `features.FilterGatewayClusterConfig` and delta clusters mutually exclusive for the gateway? Does this result in a full push (of the filtered clusters) on every change?

Reply:
In the current codepath(s), FilterGatewayClusterConfig and delta xDS are mutually exclusive. But yeah, the change as is is kinda a hack to effectively disable delta XDS for gateways when FilterGatewayClusterConfig is enabled...

I've actually been working on a slightly more complicated diff that does the filtering in the delta calculations for SE/DR.

Reply:
see https://github.com/istio/istio/pull/55826 (WIP, need to add tests)

Reply:
FilterGatewwayClusterConfig has been defaulted to false because of the issues described in https://github.com/istio/istio/issues/29131. So My attempt was to just be on par with SotW Xds